index.js

```diff
@@ -147,25 +148,32 @@ module.exports = class handlePeers extends EventEmitter {
 function readPeers (node, conn) {
   const reader = Reader()
   pull(conn, reader)
-  return new Promise((resolve, reject) => {
-    reader.read(1, (err, len) => {
+
+  return new Promise(async (resolve, reject) => {
```

Comment:
you really want avoid using async in a Promise like that, when possible. if something charshed in there and the Promise reject doesn't get called, you wont be able to see the error.  try something like this i think

```
   // read the length as bn.js instance
   const lenData = await leb.unsigned.readBn(reader)
  
    if (lenData.isZero())) {
      reader.abort()
      return resolve({})
    }
   
   return new Promise((resolve, reject) => {
     reader.read(lenData.toNumber(), (err, data) => {
      if (err) {
        return reject(err)
      }

      data = data.toString()
      const peers = JSON.parse(data)
      reader.abort()
      resolve(peers)
    })
```

Reply:
Without the `async` keyword, though, isn't it incorrect to use `await` inside the promise handler?

Reply:
right you would put `async` on the function

```
 async function readPeers (node, conn) {
    const reader = Reader()
    pull(conn, reader)
   ....
```

Reply:
D'oh. Thanks.